source/nodejs/adaptivecards/src/carousel.ts

```diff
@@ -0,0 +1,345 @@
+import { NumProperty, property, PropertyBag, Versions } from "./serialization";
+import { GlobalRegistry, ElementSingletonBehavior } from "./registry";
+import { TypeErrorType, ValidationEvent } from "./enums";
+import { Strings } from "./strings";
+import { Swiper, A11y, Autoplay, History, Keyboard, Navigation, Pagination, Scrollbar, SwiperOptions } from "swiper";
+import * as Utils from "./utils";
+import { GlobalSettings } from "./shared";
+import "swiper/css";
+import "swiper/css/pagination";
+import "swiper/css/navigation";
+
+export class CarouselPage extends Container {
+    //#region Schema
+
+    //#endregion
+
+    protected internalRender(): HTMLElement | undefined {
+        const swiperSlide: HTMLElement = document.createElement("div");
+        swiperSlide.className = this.hostConfig.makeCssClassName("swiper-slide");
+
+        const renderedElement = super.internalRender();
+        Utils.appendChild(swiperSlide, renderedElement);
+        return swiperSlide;
+    }
+
+    getForbiddenActionTypes(): ActionType[] {
+        return [ShowCardAction, ToggleVisibilityAction]
+    }
+
+    protected internalParse(source: any, context: SerializationContext) {
+        super.internalParse(source, context);
+
+        this.setShouldFallback(false);
+    }
+
+    protected shouldSerialize(context: SerializationContext): boolean {
+        return true;
+    }
+
+    getJsonTypeName(): string {
+        return "CarouselPage";
+    }
+
+    get isStandalone(): boolean {
+        return false;
+    }
+}
+
+export class Carousel extends ContainerBase {
+    //#region Schema
+
+    static readonly timerProperty = new NumProperty(Versions.v1_6, "timer", undefined);
+    @property(Carousel.timerProperty)
+    get timer(): number | undefined {
+        let timer = this.getValue(Carousel.timerProperty);
+
+        if (timer && timer < this.hostConfig.carousel.minAutoplayDelay) {
+            console.warn(Strings.errors.tooLittleTimeDelay);
+            timer = this.hostConfig.carousel.minAutoplayDelay;
+        }
+
+        return timer;
+    }
+
+    set timer(value: number | undefined) {
+        if (value && value < this.hostConfig.carousel.minAutoplayDelay) {
+            console.warn(Strings.errors.tooLittleTimeDelay);
+            this.timer = this.hostConfig.carousel.minAutoplayDelay;
+        }
+        else {
+            this.timer = value;
+        }
+    }
+
+    //#endregion
+
+    private _pages: CarouselPage[] = [];
+    private _renderedPages: CarouselPage[];
+
+    protected forbiddenChildElements(): string[] {
+        return [
+            ToggleVisibilityAction.JsonTypeName,
+            ShowCardAction.JsonTypeName,
+            "Media",
+            "ActionSet",
+            "Input.Text",
+            "Input.Date",
+            "Input.Time",
+            "Input.Number",
+            "Input.ChoiceSet",
+            "Input.Toggle",
+            ...super.forbiddenChildElements()
```

Comment:
nit: is this needed, we are parsing from top to bottom and adding a forbidden item list to the children.

Reply:
I'll take a look and remove in a subsequent PR :)